### PR TITLE
Opportunity record

### PR DESCRIPTION
**Pull request by iloveitaly**

### CODE REVIEW

File: lib/netsuite/records/opportunity.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      attr_accessor :external_id
+      attr_accessor :search_joins
+
+      def initialize(attributes = {})
+        @internal_id = attributes.delete(:internal_id) || attributes.delete(:@internal_id)
+        @external_id = attributes.delete(:external_id) || attributes.delete(:@external_id)
+        initialize_from_attributes_hash(attributes)
+      end
+
+      def self.search_class_name
```
Comment (iloveitaly) [inline lib/netsuite/records/opportunity.rb]: No need for a `search_Class_name` here... this is auto-generated. This is only required when the search name is different from the class name.

File: lib/netsuite/records/opportunity_item_list.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+module NetSuite
+  module Records
+    class OpportunityItemList
```
Comment (iloveitaly) [inline lib/netsuite/records/opportunity_item_list.rb]: This should inherit from `Support::Sublist`. Checkout this sublist as an example:

https://github.com/NetSweet/netsuite/blob/master/lib/netsuite/records/invoice_item_list.rb

File: netsuite.gemspec [hunk 3]
```diff
@@ -15,7 +15,7 @@ Gem::Specification.new do |gem|
   gem.require_paths = ['lib']
   gem.version       = Netsuite::VERSION
 
-  gem.add_dependency 'savon', '~> 2.3.0'
+  gem.add_dependency 'savon', '>= 2.3.0'
```
Comment (iloveitaly) [inline netsuite.gemspec]: There is a PR for this, let's leave this out of this PR if possible.

https://github.com/NetSweet/netsuite/pull/172

File: lib/netsuite/records/opportunity_item_list.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+module NetSuite
+  module Records
+    class OpportunityItemList < Support::Sublist
```
Comment (iloveitaly) [inline lib/netsuite/records/opportunity_item_list.rb]: @jheth `OpportunityItemList` shouldn't contain any logic aside a namespace definition and a `sublist` call. Ex: https://github.com/NetSweet/netsuite/blob/de60d36403587b9d255b9d85ec1d4651af0281cc/lib/netsuite/records/invoice_item_list.rb#L4-L6

**Comment by iloveitaly** [comment]
Looking good! One additional comment: any new records should be added to the basic record spec:

https://github.com/NetSweet/netsuite/blob/master/spec/netsuite/records/basic_record_spec.rb

**Comment by jheth** [comment]
Thanks @iloveitaly.  Just added to basic_record_spec.  Should I add an opportunity_spec.rb as well?

**Comment by iloveitaly** [comment]
> Should I add an opportunity_spec.rb as well?

I've been bearish on individual specs for records. I'm fine with just a `basic_record_spec` entry.

If there is no unique record functionality, they add a lot of maintinance weight with very little value. What do you think?

Thanks for the PR!

**Comment by iloveitaly** [comment]
Merged with https://github.com/NetSweet/netsuite/commit/371baf96794c5209357f6f5e2100868f000fc9c9

**Pull request closed without merging**